File: backend/src/main/java/com/culinaryheaven/domain/comment/dto/response/CommentResponse.java
package com.culinaryheaven.domain.comment.dto.response;

import com.culinaryheaven.domain.comment.domain.Comment;
import com.culinaryheaven.domain.user.dto.response.UserInfoResponse;
import io.swagger.v3.oas.annotations.media.Schema;

import java.time.LocalDateTime;

public record CommentResponse(

        @Schema(description = "댓글 ID")
        Long id,

        @Schema(description = "댓글 내용")
        String content,

        @Schema(description = "댓글 작성자")
        UserInfoResponse userInfo,

        @Schema(description = "댓글 작성 시간")
        LocalDateTime createdAt
) {

        public static CommentResponse of(Comment comment) {
                return new CommentResponse(
                        comment.getId(),
                        comment.getContent(),
                        UserInfoResponse.of(comment.getUser()),
                        comment.getCreatedAt()
                );
        }

}

File: android/data/src/main/java/com/jae464/data/remote/model/response/comment/CommentResponse.kt
package com.jae464.data.remote.model.response.comment

import com.jae464.data.remote.model.response.user.UserInfoResponse
import com.jae464.data.remote.model.response.user.toDomain
import com.jae464.domain.model.Comment
import kotlinx.serialization.Serializable
import java.time.LocalDateTime

@Serializable
data class CommentResponse(
    val id: Long,
    val content: String,
    val createdAt: String,
    val userInfo: UserInfoResponse
)

fun CommentResponse.toDomain() = Comment(
    id = id,
    content = content,
    createdAt = LocalDateTime.parse(createdAt),
    userInfo = userInfo.toDomain()
)